README.rst

```diff
@@ -1506,79 +1506,366 @@ Get command details
+
+The following example demonstrates creating related location
+object in a single request.
+
+.. code-block:: json
+
+    {
+        "location": {
+            "name": "Via del Corso",
+            "address": "Via del Corso, Roma, Italia",
+            "geometry": "POINT (12.512124 41.898903)",
+            "type": "outdoor",
+        }
+    }
+
+.. code-block:: text
+
+    curl -X POST \
+        http://127.0.0.1:8000/api/v1/controller/device/8a85cc23-bad5-4c7e-b9f4-ffe298defb5c/location/ \
+        -H 'content-type: application/json' \
+        -d '{
+                "location": {
+                    "name": "Via del Corso",
+                    "address": "Via del Corso, Roma, Italia",
+                    "geometry": "POINT (12.512124 41.898903)",
+                    "type": "outdoor"
+                }
+            }'
+
+Similarly, you can create ``Floorplan`` object with the same request.
+But, note that a ``FloorPlan`` can be added to ``DeviceLocation`` only
+if the related ``Location`` object defines an indoor location. The example
+below demonstrates creating both ``Location`` and ``FloorPlan`` objects.
+
+.. code-block:: text
+
+    // This is not a valid JSON object. The JSON format is
+    // only used for showing available fields.
+    {
+        "location.name": "Via del Corso",
+        "location.address": "Via del Corso, Roma, Italia",
+        "location.geometry": "POINT (12.512124 41.898903)",
+        "location.type": "outdoor",
+        "floorplan.floor": 1,
+        "floorplan.image": floorplan.png,
+    }
+
+.. code-block:: text
+
+    curl -X POST \
+        http://127.0.0.1:8000/api/v1/controller/device/8a85cc23-bad5-4c7e-b9f4-ffe298defb5c/location/ \
+        -H 'content-type: multipart/form-data; boundary=----WebKitFormBoundary7MA4YWxkTrZu0gW' \
+        -F 'location.name=Via del Corso' \
+        -F 'location.address=Via del Corso, Roma, Italia' \
+        -F 'location.geometry=POINT (12.512124 41.898903)' \
+        -F location.type=indoor \
+        -F floorplan.floor=1 \
+        -F 'floorplan.image=@floorplan.png'
+
+**Note:** The request in above example uses ``multipart content-type``
+for uploading floorplan image.
+
+You can also use an existing ``Location`` object and create a new
+floorplan for that location using this endpoint.
+
+.. code-block:: text
+
+    // This is not a valid JSON object. The JSON format is
+    // only used for showing available fields.
+    {
+        "location": "f0cb5762-3711-4791-95b6-c2f6656249fa",
+        "floorplan.floor": 1,
+        "floorplan.image": floorplan.png
+    }
+
+.. code-block:: text
+
+    curl -X POST \
+        http://127.0.0.1:8000/api/v1/controller/device/8a85cc23-bad5-4c7e-b9f4-ffe298defb5c/location/ \
+        -H 'content-type: multipart/form-data; boundary=----WebKitFormBoundary7MA4YWxkTrZu0gW' \
+        -F location=f0cb5762-3711-4791-95b6-c2f6656249fa \
+        -F floorplan.floor=1 \
+        -F 'floorplan.image=@floorplan.png'
+
```

Comment:
Is it mentioned anywhere in this endpoint that it can be used to update/change too?